pkg/restore/client.go

```diff
@@ -455,6 +466,159 @@ func (rc *Client) ValidateChecksum(
 	return nil
 }
 
+const (
+	restoreLabelKey   = "exclusive"
+	restoreLabelValue = "restore"
+)
+
+// LoadRestoreStores loads the stores used to restore data.
+func (rc *Client) LoadRestoreStores(ctx context.Context) error {
+	if !rc.isOnline {
+		return nil
+	}
+
+	stores, err := rc.pdClient.GetAllStores(ctx)
+	if err != nil {
+		return err
+	}
+	for _, s := range stores {
+		if s.GetState() != metapb.StoreState_Up {
+			continue
+		}
+		for _, l := range s.GetLabels() {
+			if l.GetKey() == restoreLabelKey && l.GetValue() == restoreLabelValue {
+				rc.restoreStores = append(rc.restoreStores, s.GetId())
+				break
+			}
+		}
+	}
+	log.Info("load restore stores", zap.Uint64s("store-ids", rc.restoreStores))
+	return nil
+}
+
+// ResetRestoreLabels removes the exclusive labels of the restore stores.
+func (rc *Client) ResetRestoreLabels(ctx context.Context) error {
+	if !rc.isOnline {
+		return nil
+	}
+	log.Info("start reseting store labels")
+	return rc.toolClient.SetStoresLabel(ctx, rc.restoreStores, restoreLabelKey, "")
+}
+
+// SetupPlacementRules sets rules for the tables' regions.
+func (rc *Client) SetupPlacementRules(ctx context.Context, tables []*model.TableInfo) error {
+	if !rc.isOnline || len(rc.restoreStores) == 0 {
```

Comment:
mayby we shoud refuse to start restore online if `len(rc.restoreStores) == 0`
if there is no proper restore stores, force restore may affect online cluster.

Reply:
It was designed to use [rateLimit](https://github.com/pingcap/br/pull/100) to take care of this problem. If not to provide new stores, user need to set to lower rate limit to reduce performance impact.